src/ethereum_test_tools/filling/fill.py

```diff
@@ -34,6 +34,20 @@ def fill_test(
         eips=eips,
     )
 
+    fcu_version: int | None = None
+    if not test_spec.base_test_config.disable_hive:
+        last_valid_block = next(
```

Comment:
I think it would be a bit cleaner to get the last valid number and timestamp from the `make_blocks` call directly, so we don't have to iterate it again here, and potentially cause issues if we change the logic for some reason inside of the spec generators.